app/models/concerns/employed_income.rb

```diff
@@ -55,6 +55,14 @@ def employed_income_payment_types
     end
   end
 
+  def require_client_in_armed_forces?
+    client_employed_only?
+  end
+
+  def require_partner_in_armed_forces?
+    partner_employed_only? && employment_status.exclude?(EmploymentStatus::SELF_EMPLOYED.to_s)
```

Comment:
I think `employment_status.exclude?(EmploymentStatus::SELF_EMPLOYED.to_s)` can be replaced with `client_employed_only?`